Data/HashMap/Base.hs

```diff
@@ -448,6 +455,43 @@ lookup k0 m0 = go h0 k0 0 m0
         | otherwise = Nothing
 {-# INLINABLE lookup #-}
 
+-- Internal helper for lookup. This version takes the precomputed hash so
+-- that functions that make multiple calls to lookup and related functions
+-- (insert, delete) only need to calculate the hash once.
+--
+-- It is used by 'alterF' so that hash computation and key comparison only needs
+-- to be performed once. With this information you can use the more optimized
+-- versions of insert ('insertNewKey', 'insertKeyExists') and delete
+-- ('deleteKeyExists')
+--
+-- Outcomes:
+--   Key not in map           => Absent
+--   Key in map, no collision => Alone v
+--   Key in map, collision    => Collide v position
+lookupRecordCollision
+  :: (Eq k, Hashable k)
+  => Hash -> k -> Int -> HashMap k v -> LookupRes v
```

Comment:
What is the `Int` argument? Why a `Hashable` constraint?

Reply:
Int argument is needed for the BitMapIndexed and Full case. Hashable constraint removed.